Code/Tools/AssetProcessor/AssetBuilderSDK/AssetBuilderSDK/SerializationDependencies.cpp

```diff
@@ -24,16 +24,16 @@ namespace AssetBuilderSDK
         {
             return false;
         }
-        if (classData->m_typeId == AZ::GetAssetClassId())
+        const auto isAssetClass = [&serializeContext, classData]
         {
-            auto* asset = reinterpret_cast<AZ::Data::Asset<AZ::Data::AssetData>*>(instancePointer);
-
-            if (asset->GetId().IsValid())
+            if (classData->m_typeId == AZ::GetAssetClassId() && classData->m_version == 1)
```

Comment:
This needs to be `classData->m_version <=2` as version 2 `AZ::Data::Asset` class serialization writes out the AssetClassId